packages/pigeon/example/app/linux/messages.g.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v11.0.1), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+#ifndef PIGEON_MESSAGES_G_H_
+#define PIGEON_MESSAGES_G_H_
+
+#include <flutter_linux/flutter_linux.h>
+
+G_BEGIN_DECLS
+
+typedef enum { MY_CODE_ONE = 0, MY_CODE_TWO = 1 } MyCode;
+
+G_DECLARE_FINAL_TYPE(MyMessageData, my_message_data, MY, MESSAGE_DATA, GObject)
+
+MyMessageData* my_message_data_new(MyCode code, FlValue* data);
+
+MyMessageData* my_message_data_new_full(const gchar* name,
+                                        const gchar* description, MyCode code,
+                                        FlValue* data);
+
+const gchar* my_message_data_get_name(MyMessageData* object);
+
+const gchar* my_message_data_get_description(MyMessageData* object);
+
+MyCode my_message_data_get_code(MyMessageData* object);
+
+FlValue* my_message_data_get_data(MyMessageData* object);
+
+G_DECLARE_FINAL_TYPE(MyExampleHostApi, my_example_host_api, MY,
+                     EXAMPLE_HOST_API, GObject)
+
+typedef struct {
+  void (*get_host_language)(
+      MyExampleHostApi* object,
+      FlBasicMessageChannelResponseHandle* response_handle, gpointer user_data);
+  void (*add)(MyExampleHostApi* object,
+              FlBasicMessageChannelResponseHandle* response_handle, int64_t a,
+              int64_t b, gpointer user_data);
+  void (*send_message)(MyExampleHostApi* object,
+                       FlBasicMessageChannelResponseHandle* response_handle,
+                       MyMessageData* message, gpointer user_data);
+} MyExampleHostApiVTable;
+
+MyExampleHostApi* my_example_host_api_new(FlBinaryMessenger* messenger,
+                                          const MyExampleHostApiVTable* vtable,
+                                          gpointer user_data,
+                                          GDestroyNotify user_data_free_func);
+
+gboolean my_example_host_api_respond_get_host_language(
+    MyExampleHostApi* object,
+    FlBasicMessageChannelResponseHandle* response_handle, const gchar* result,
+    GError** error);
+
+gboolean my_example_host_api_respond_error_get_host_language(
```

Comment:
If we do custom response handles, we could also make it contain the channel, and then we wouldn't need a different error callback for each method; it could be generic.

And in fact we could do a non-error return per used return type, instead of per method. E.g., if the type hierarchy were something like `PigeonResponseHandle` -> `PigeonResponseHandleBool`, `PigeonResponseHandleInt`, etc. (where we generate only the ones the API actually needs), then the error response could take `PigeonResponseHandle` and then we could have `my_example_host_api_respond_bool(..., PigeonResponseHandleBool* response_handle, bool result, ...) and so on.

What do you think?

Reply:
Since the code is generated, it seems better to me to have many functions that do more type checking and thus reduce the risk of using the API wrong.